### ISSUE TITLE
Social icon widget miss spacing between icons if HTML minification is used on a site

### ISSUE THREAD

**Issue by vovkasolovev** [elementor/elementor#13279]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

Social icon widget miss spacing between icons if HTML minification is used on a site.

## Steps to reproduce

1. Use HTML minification with Optimize HTML Code option on.
2. Add Social icon widget to the template and set spacing between social icons.
3. Preview the page to see that social icons are sticked together.

After HTML get minified and html is flattened there is no any space character inside widget HTML code and CSS rule
`word-spacing: var(--grid-column-gap);`
does not applied for inline block elements.

**Comment by Zach-Toth** [comment]
Same issue here. Disabling HTML options in Autoptomize solves the issue, as does deactivating Autoptomize, but I need HTML minification for performance

**Comment by Aviad-Herman** [comment]
Hey @vovkasolovev ,
Thanks for reporting this.
It seems like the problem is with the HTML optimization plugin. Maybe try another one like W3 Cache?

Another solution is to add custom CSS to the social icons widget like so:
![image](https://user-images.githubusercontent.com/73067562/102825089-f46cf680-4392-11eb-9874-4f6bbc97688a.png)

**Comment by vovkasolovev** [comment]
Why did you close this issue? 
This is not a problem with optimization plugins, but with CSS. Any optimization plugin just cuts out the extra spaces, after that your code doesn't work, because the alignment is based on the width of the space characters (word-spacing) in HTML. **It is not correct to use HTML for styling**.

Why suggest using custom CSS instead of fixing what this code fixes?

**Comment by Aviad-Herman** [comment]
I'm sorry but when minimizing html using W3 Total Cache it works fine which might indicate that the problem is with Autoptimize. In case you want to keep using Autoptimize I gave you a hack you can use. I hope this helps.

**Comment by vovkasolovev** [comment]
Please, forget about Autoptimize and open issue. Issue is about using "word-spacing" with is wrong and can't be fixed with your hack.

**Comment by vovkasolovev** [comment]
Thank you!

**Comment by realstephenzhao** [comment]
Any progress? This issue also happens with SG Optimizer(from Siteground Hosting).

**Comment by mwaterhouse** [comment]
The workaround is useful but this does need fixing properly.
WP-Optimize (1m+ active installs) [even recommend](https://wordpress.org/support/topic/this-plugin-conflicts-with-elementor-social-media-icons/#post-13632226) turning off HTML minification as a 'solution' because _"The way Elementor handles spacing of the social icons is not compatible with HTML minification."_
The problem is with Elementor.

**Comment by black-eye** [comment]
+1

**Comment by wwdes** [comment]
+1

**Comment by jormeer** [comment]
+1 spacing option doesnt work

**Comment by jormeer** [comment]
I am confused should this be fixed now in pro 3.3? Since it looks like it isnt?

**Comment by gerardreches** [comment]
+1, also happening with the LiteSpeed Cache plugin when HTML minification is turned on.

More people experiencing this issue here: [Elementor support on WordPress.org](https://wordpress.org/support/topic/elementor-social-icons-do-not-display-space-when-html-minify-option-is-activated/page/2/#post-14535322)

@jormeer Same here.


```
@shilo-ey shilo-ey mentioned this issue on 5 May
📣 Elementor Pro v3.3 Beta 5 Release 📣 #14827 
```

Why cannot I find this issue mentioned in that post while here it says that it was mentioned?

**Comment by gerardreches** [comment]
Adding the following code to your widget's Custom CSS box will solve the problem until we get it fixed in a future update:

```
selector .elementor-grid .elementor-grid-item:not(:last-child) {
    margin-right: var(--grid-column-gap);
}
selector .elementor-grid{
    word-spacing: initial;
}
```

It doesn't use a fixed value, so you will still be able to adjust the gap between buttons from the Style settings of the widget. The second part of this CSS code is to avoid getting the gap twice if HTML modification is disabled.

Easy solution. Yet the Elementor team doesn't seem to want to add it by themselves.

**We need this to be fixed by the plugin itself, not by adding our own custom CSS.**

**Comment by DorShahar** [comment]
Hi Everyone
We want to make sure we released a stable fix for this bug.
As marked above, this fix been merged to the next Core version - v3.3.0.

We encourage you to test in the Beta version and report us if there is still an issue (please report on Beta thread #15336 ).

Thank you.

**Comment by DorShahar** [comment]
Hi everyone! 
This issue has been resolved in Elementor v3.3.0

Feel free to update

Thanks!
